### PR TITLE
fix: ensure content field present on tool call result messages

### LINKED ISSUE #3198: 🐛 Bug: Backend Actions zod validation issue in 1.51.3
### ♻️ Reproduction Steps

I have made a [repro branch](https://github.com/SnowLeopard-AI/snowy-examples/tree/ll/copilotkit_integration/quickstart/copilot-kit) if that is helpful, but this error reproduces from following your hello world example exactly.

1. Follow the "[direct to LLM (self-hosted)](https://docs.copilotkit.ai/direct-to-llm/guides/quickstart?copilot-hosting=self-hosted)" and "[Backend Action (typescript)](https://docs.copilotkit.ai/direct-to-llm/guides/backend-actions/typescript-backend-actions)" docs

2. You will end up with a route.ts which looks like
```
import {
    CopilotRuntime,
    OpenAIAdapter,
    copilotRuntimeNextJSAppRouterEndpoint,
} from '@copilotkit/runtime';

import { NextRequest } from 'next/server';


let openAIAdapter = new OpenAIAdapter({model: "gpt-5-mini"});
const runtime = new CopilotRuntime({
    actions: ({properties, url}) => {
        return [
            {
                name: "fetchNameForUserId",
                description: "Fetches user name from the database for a given ID.",
                parameters: [
                    {
                        name: "userId",
                        type: "string",
                        description: "The ID of the user to fetch data for.",
                        required: true,
                    },
                ],
                handler: async ({userId}: {userId: string}) => {
                    return {
                        name: "Darth Doe",
                    };
                },
            },
        ]
    }
});

export const POST = async (req: NextRequest) => {
    const { handleRequest } = copilotRuntimeNextJSAppRouterEndpoint({
        runtime,
        serviceAdapter: openAIAdapter,
        endpoint: '/api/copilotkit',
    });

    return handleRequest(req);
};
```

3. Run with 'npm run dev'
4. Go to dev address and ask agent "what is the name of user 'abcd'"

### ✅ Expected Behavior

The tool call is made and "Darth Doe" is returned

### ❌ Actual Behavior

5. Zod Error 
> Note this zod error is on the frontend, validating the argument generation produced by the backend! **The arguments are valid, but the object that copiliotkit wraps them in are not**

Image
Image

### 𝌚 CopilotKit Version

```shell
npm ls | grep "@copilotkit"
├── @copilotkit/react-core@1.51.3
├── @copilotkit/react-ui@1.51.3
├── @copilotkit/runtime@1.51.3
```

### 📄 Logs (Optional)

```shell
## Error Type
Console ZodError

## Error Message
[
  {
    "code": "invalid_type",
    "expected": "string",
    "received": "undefined",
    "path": [
      "content"
    ],
    "message": "Required"
  }
]

Next.js version: 16.1.5 (Turbopack)


/Users/luke/Downloads/copilotkit-events-1770853621928.json
```

### PR DESCRIPTION
**Pull request by jpr5**
## Summary

- Backend action handler was constructing tool call result messages without the required `content` field, causing Zod validation failures
- The `content` field is required by the message schema but was omitted when wrapping tool results
- Added `content` field to tool result message construction, with warning logs for undefined results

Fixes #3198

## Test plan

- [x] Unit tests verifying tool result messages include `content` field
- [x] Tests for edge cases: undefined results, missing handlers
- [x] Full package test suite passes
- [x] Build passes

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: packages/v1/runtime/src/lib/runtime/copilot-runtime.ts [hunk 1]
```diff
@@ -454,7 +454,16 @@ export class CopilotRuntime<const T extends Parameter[] | [] = []> {
         name: action.name,
         description: action.description || "",
         parameters: zodSchema,
-        execute: () => Promise.resolve(),
+        execute: async (args: any) => {
```
Comment (AlemTuzlak) [inline packages/v1/runtime/src/lib/runtime/copilot-runtime.ts:457]: can we use unknown here? I hate the fact that are codebase is riddled with "any" which then implicitly tells coding agents that's cool to do

File: packages/v2/agent/src/__tests__/tool-result-content.test.ts [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+        toolResult("call1", "backendAction", undefined),
+        finish(),
+      ]) as any,
+    );
+
+    const events = await collectEvents(agent["run"](baseInput));
+
+    const resultEvent = events.find(
+      (e: any) => e.type === EventType.TOOL_CALL_RESULT,
+    ) as any;
```
Comment (AlemTuzlak) [inline packages/v2/agent/src/__tests__/tool-result-content.test.ts:71]: look at these tests, as any as any as any 😢

File: packages/v2/agent/src/index.ts [hunk 3]
```diff
@@ -1116,12 +1116,21 @@ export class BuiltInAgent extends AbstractAgent {
                 }
 
                 // Always emit the tool result event for the LLM
+                // JSON.stringify(undefined) returns undefined (not a string),
+                // so we fall back to "" to satisfy the Zod schema requirement
+                // that content is always a string.
+                const serialized = JSON.stringify(toolResult);
```
Comment (AlemTuzlak) [inline packages/v2/agent/src/index.ts:1122]: This is a bug waiting to happen, JSON.stringify throws on circular deps (unlikely) and BigInt present in the structure (which means this would throw if a BigInt ends up as a result)

**Review by AlemTuzlak — CHANGES_REQUESTED** [review]

**Comment by jpr5** [comment]
Closing — the v1/v2 package structure this PR targets no longer exists (repo restructured to flat packages/). 

The underlying bug (missing `content` field on tool call result messages) is fixed in the current v2 agent infrastructure — `packages/runtime/src/agent/index.ts:1334` always sets `content: JSON.stringify(toolResult)` on `ToolCallResultEvent`. Closing #3198 as well.
